File: plugins/processman/testdata/http_server.py
#!/usr/bin/env python3
"""
Python HTTP 服务器 - 用于测试 HTTP 服务进程管理
提供健康检查和简单 API 端点
"""
import json
import os
import sys
from http.server import HTTPServer, BaseHTTPRequestHandler

PORT = int(os.environ.get("HTTP_PORT", "9100"))

class SimpleHandler(BaseHTTPRequestHandler):
    def log_message(self, format, *args):
        # 输出到 stdout 便于日志桥接测试
        print(f"[HTTP] {self.address_string()} - {args[0]}")
    
    def do_GET(self):
        if self.path == "/health":
            self._json_response(200, {"status": "healthy", "pid": os.getpid()})
        elif self.path == "/info":
            self._json_response(200, {
                "service": "test-http-server",
                "pid": os.getpid(),
                "port": PORT,
                "env": dict(os.environ)
            })
        elif self.path == "/shutdown":
            self._json_response(200, {"message": "shutting down"})
            # 延迟关闭，让响应先发出去
            import threading
            threading.Timer(0.1, lambda: os._exit(0)).start()
        else:
            self._json_response(404, {"error": "not found"})
    
    def _json_response(self, status, data):
        self.send_response(status)
        self.send_header("Content-Type", "application/json")
        self.end_headers()
        self.wfile.write(json.dumps(data).encode())

def main():
    print(f"[HTTP] Starting server on port {PORT}, PID: {os.getpid()}")
    sys.stdout.flush()
    
    server = HTTPServer(("127.0.0.1", PORT), SimpleHandler)
    print(f"[HTTP] Server ready at http://127.0.0.1:{PORT}")
    sys.stdout.flush()
    
    try:
        server.serve_forever()
    except KeyboardInterrupt:
        print("[HTTP] Received shutdown signal")
    finally:
        server.server_close()
        print("[HTTP] Server stopped")

if __name__ == "__main__":
    main()

File: plugins/processman/testdata/http_server.js
#!/usr/bin/env node
/**
 * Node.js HTTP 服务器 - 用于测试 HTTP 服务进程管理
 * 提供健康检查和简单 API 端点
 */

const http = require('http');

const PORT = parseInt(process.env.HTTP_PORT || '9101', 10);

const server = http.createServer((req, res) => {
    const url = new URL(req.url, `http://localhost:${PORT}`);
    console.log(`[HTTP] ${req.method} ${url.pathname}`);
    
    res.setHeader('Content-Type', 'application/json');
    
    if (req.method === 'GET') {
        if (url.pathname === '/health') {
            res.writeHead(200);
            res.end(JSON.stringify({ status: 'healthy', pid: process.pid, runtime: 'node' }));
        } else if (url.pathname === '/info') {
            res.writeHead(200);
            res.end(JSON.stringify({
                service: 'test-http-server',
                pid: process.pid,
                port: PORT,
                runtime: process.version,
                env: process.env
            }));
        } else if (url.pathname === '/shutdown') {
            res.writeHead(200);
            res.end(JSON.stringify({ message: 'shutting down' }));
            setTimeout(() => process.exit(0), 100);
        } else {
            res.writeHead(404);
            res.end(JSON.stringify({ error: 'not found' }));
        }
    } else {
        res.writeHead(405);
        res.end(JSON.stringify({ error: 'method not allowed' }));
    }
});

server.listen(PORT, '127.0.0.1', () => {
    console.log(`[HTTP] Starting server on port ${PORT}, PID: ${process.pid}`);
    console.log(`[HTTP] Server ready at http://127.0.0.1:${PORT}`);
});

// 处理信号
process.on('SIGTERM', () => {
    console.log('[HTTP] Received SIGTERM');
    server.close(() => {
        console.log('[HTTP] Server stopped');
        process.exit(0);
    });
});

process.on('SIGINT', () => {
    console.log('[HTTP] Received SIGINT');
    server.close(() => {
        console.log('[HTTP] Server stopped');
        process.exit(0);
    });
});
